appinventor/buildserver/src/com/google/appinventor/buildserver/resources/runtime.scm

```diff
@@ -434,12 +443,37 @@
                          registeredComponentName eventName)
                        #f))))
 
+       ;; For the HandlesEventDispatching interface
+       (define (dispatchMultiEvent eventName :: java.lang.String
```

Comment:
The second block is like a normal event handler except you also have to give it the component that you want to use it for. So these two blocks behave the same (hence the error):

![screen shot 2015-04-12 at 1 00 31 pm](https://cloud.githubusercontent.com/assets/4552296/7106576/a12e2e68-e113-11e4-8cbc-f5cea7d03e7f.png)

The third block handles events for all instances of specific component and returns the instance that actually triggered the event.